python/pyarrow/_s3fs.pyx

```diff
@@ -157,11 +157,16 @@ cdef class S3FileSystem(FileSystem):
                                         'port': 8020, 'username': 'username',
                                         'password': 'password'})
     allow_bucket_creation : bool, default False
-        Whether to allow CreateDir at the bucket-level. This option may also be 
+        Whether to allow CreateDir at the bucket-level. This option may also be
         passed in a URI query parameter.
     allow_bucket_deletion : bool, default False
-        Whether to allow DeleteDir at the bucket-level. This option may also be 
+        Whether to allow DeleteDir at the bucket-level. This option may also be
         passed in a URI query parameter.
+    retry_strategy : str, default None
+        The name of the retry strategy to use with S3.  Valid values are
+        "aws_standard" and "aws_default". Ignore with None.
+    retry_max_attempts : int, default 3
+        The maximum number of attempts to pass to AWS retry strategies.
```

Comment:
I think it is a bad design choice to use two separate arguments for this.
I would favour a single argument, perhaps together with more structured objects, e.g.:
```python
class AwsDefaultRetryStrategy(object):
    def __init__(self, max_attempts=3):
        self.max_attempts = max_attempts
```